kalite/coachreports/features/steps/coachreports.py

```diff
@@ -17,5 +36,181 @@ def step_impl(context):
+@then(u"there should be three learner rows displayed")
+def impl(context):
+    # Wait for appearance
+    find_css_class_with_wait(context, "user-data-row")
+    # Find all the rows
+    rows = context.browser.find_elements_by_css_selector(".user-data-row")
+    assert len(rows) == 3, "Incorrect number of user rows in tabular report"
+
+@when(u"I select the preferred group")
+def impl(context):
+    dropdown = Select(find_id_with_wait(context, "group-select"))
+    dropdown.select_by_visible_text("0")
+    # For some reason this only triggers properly if you defocus or press enter after the select
+    dropdown = find_id_with_wait(context, "group-select")
+    dropdown.send_keys(Keys.RETURN)
+
+@given(u"I am on the tabular report")
+def impl(context):
+    context.execute_steps(u"""
+        Given I am on the coach report
+        When I click on the Show Tabular Report button
+        Then I should see the tabular report
+        """)
+
+@given(u"there are three learners")
+def impl(context):
+    for f in FacilityUser.objects.all():
+        f.soft_delete()
+    name = 0
+    while FacilityUser.objects.count() < 3:
+        context.facility = CreateStudentMixin.create_student(username=str(name))
+        name += 1
+
+@given(u'the "{learner}" "{progress_verbs}" an "{exercise}"')
+def impl(context, learner, progress_verbs, exercise):
+    user = CreateStudentMixin.create_student(username=learner)
+    if not progress_verbs == "not started":
+        log, created = ExerciseLog.objects.get_or_create(exercise_id=exercise, user=user)
+        if progress_verbs == "completed":
+            log.streak_progress = 100
+            log.attempts = 15
+        elif progress_verbs == "attempted":
+            log.streak_progress = 50
+            log.attempts = 10
+        elif progress_verbs == "struggling":
+            log.streak_progress = 30
+            log.attempts = 25
+        for i in range(0, log.attempts):
+            attempt_log, created = AttemptLog.objects.get_or_create(exercise_id=exercise, user=user, seed=i, timestamp=datetime.datetime.now())
+        log.save()
+
+@then(u'I should see the "{exercise}" marked for "{learner}" as "{progress_text}" and coloured "{progress_colour}"')
+def impl(context, exercise, learner, progress_text, progress_colour):
+    if progress_text == u"None":
+        progress_text = u""
+
+    user = FacilityUser.objects.get(username=learner)
+    data_row = find_id_with_wait(context, user.id)
+    cell = data_row.find_element_by_css_selector("td[value={exercise}]".format(exercise=exercise))
+    assert cell.text == progress_text, "Progress text for {learner}, on {exercise} incorrect".format(learner=learner, exercise=exercise)
+    assert rgba_to_hex(cell.value_of_css_property("background-color")) == colour_legend[progress_colour]
+
+@when(u"I click on the learner name")
+def impl(context):
+    student_name = find_css_class_with_wait(context, "student-name")
+    student_name.find_element_by_tag_name("a").click()
```

Comment:
Is the click here supposed to trigger a page load? Then you should use [click_and_wait_for_page_load](https://github.com/learningequality/ka-lite/blob/develop/kalite/testing/behave_helpers.py#L56).